cpp/src/fil/fil.cu

```diff
@@ -461,6 +500,44 @@ void tl2fil_dense(std::vector<dense_node_t>* pnodes, forest_params_t* params,
   }
 }
 
+void tl2fil_dense_multi(std::vector<dense_node_t>* pnodes, forest_params_t* params,
```

Comment:
Again, looks like it has a lot of common code with `tl2fil_dense` above. Note that it will also be required to have `tl2fil_sparse_multi` if we want to support importing multiple models into FIL.

I think this is another good reason to merge treelite models outside of FIL.

Reply:
Yes, it does, I took the tl2fil_dense code and updated the required steps to be used for concatenating the trees.